lld/ELF/InputFiles.h

```diff
@@ -335,6 +339,12 @@ class BitcodeFile : public InputFile {
   void postParse();
   std::unique_ptr<llvm::lto::InputFile> obj;
   std::vector<bool> keptComdats;
+
+private:
+  StringRef saved_symbol(const char *S);
```

Comment:
I realize Teresa has already suggested you to use `unique_saver().save()` directly.  Even if you keep these `saved_symbol`, the one that takes `cons char *` should be redundant because we can `StringRef` has an implicit constructor from `const char *`.

The same applies to the one below that takes `const std::string &`.

Reply:
I actually take back what I said above.  When you have `std::string`, it can be implicitly cast to either `StringRef` or `const Twine &`, causing ambiguity.  The same story probably applies to `const char *`.

Reply:
Use savers in linker context and undo header changes.